xdsl/transforms/experimental/ConvertStencilToLLMLIR.py

```diff
@@ -160,6 +160,15 @@ def match_and_rewrite(self, op: LoadOp, rewriter: PatternRewriter, /):
 
         verify_load_bounds(cast, op)
 
+        assert op.lb and op.ub
+        assert isinstance(op.res.typ, TempType)
+        res_typ: TempType[Attribute] = op.res.typ
+
+        op.res.typ = TempType.from_shape(
```

Comment:
Yeah, it seems like we should go through the rewriter, replacing the whole op with the new parameters

Reply:
This modifies the value's type, yes, and it's used by uses, so everything's updated properly, or am I missing something? Why change a whole operation to update its returned value's type?

Reply:
As I said, it's fine if you are confident that it works. It's jsut that it is dangerous as you are changing IR in a lot of other places. Doing it through the rewriter could give you more guarantees that you don't break something accidentalyl.

Reply:
Just not sure we want to do this through diretly accessing the data strcuture and not with some kind of wrapper around it. But it's fine for te time being.

Reply:
Yeah, we could add a wrapper around it at some point! But not here!

Reply:
Nice, then I'll use the experimental TODO joker here.
I think it's even an opportunity in some cases like here, where we make sure to revisit this later with a more mature  API, instead of just compromising on something the would probably not be updated for a while!